### PR TITLE
fix(iroh): handle rpc args in any position

### LINKED ISSUE #1639: cli: rpc-port argument is confusing and partially broken
Currently when running `iroh --rpc-port PORT start` it will be ignored, but for all other commands this is the position it has to be. For it to be accepted `iroh start --rpc-port PORT` has to be used.

### PR DESCRIPTION
**Pull request by dignifiedquire**
It is now possible to do any of these 

```
> iroh start --rpc-port 1234
> iroh --rpc-port 1234 start
> iroh console --rpc-port 1234
> iroh --rpc-port 1234 console
```

Closes #1639

### CODE REVIEW

File: iroh/src/commands/node.rs [hunk 1]
```diff
@@ -28,7 +28,7 @@ use super::{BlobAddOptions, MAX_RPC_CONNECTIONS, MAX_RPC_STREAMS};
 #[derive(Debug)]
 pub struct StartOptions {
     pub addr: SocketAddr,
-    pub rpc_port: RpcPort,
+    pub rpc_port: u16,
```
Comment (rklaehn) [inline iroh/src/commands/node.rs:31]: So we can't start with rpc port disabled anymore?

Comment (dignifiedquire) [reply iroh/src/commands/node.rs:29]: For now I have removed the ability, yes. Lets work on getting the portless version for rpc supported soon to improve on this.

**Comment by Arqu** [comment]
Something doesn't sit right, netsim works with this though it doesn't pass in `--rpc-port` at start.

**Review by Arqu — APPROVED** [review]
Other than it magically working when it maybe shouldn't, LGTM.

**Pull request merged**